ta4j/src/main/java/org/ta4j/core/indicators/pivotpoints/DeMarkPivotPointIndicator.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 Marc de Verdelhan & respective authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.pivotpoints;
+
+import org.ta4j.core.Decimal;
+import org.ta4j.core.Tick;
+import org.ta4j.core.TimeSeries;
+import org.ta4j.core.indicators.RecursiveCachedIndicator;
+
+import java.time.temporal.IsoFields;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * DeMark Pivot Point indicator.
+ * <p>
+ * @author team172011(Simon-Justus Wimmer), 11.10.2017
+ * @see <a href="http://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:pivot_points">chart_school: pivotpoints</a>
+ */
+public class DeMarkPivotPointIndicator extends RecursiveCachedIndicator<Decimal> {
+
+    public static final int PIVOT_TIME_LEVEL_ID_TICKBASED = 0;
+    public static final int PIVOT_TIME_LEVEL_ID_DAY = 1; // 1-, 5-, 10- and 15-minute charts use the prior day's high, low and close.
+    public static final int PIVOT_TIME_LEVEL_ID_WEEK = 2; // 30- 60- and 120-minute charts use the prior week's high, low, and close.
+    public static final int PIVOT_TIME_LEVEL_ID_MONTH = 3; // Pivot Points for daily charts use the prior month's data.
+    public static final int PIVOT_TIME_LEVEL_ID_YEAR = 4; // Pivot Points for weekly and monthly charts use the prior year's data
```

Comment:
please use enmu:

```
public enum TimeLevel{
tickbased(0), day(1), week(2), month(3), year(4);
..
}

public DeMarkPivotPointIndicator(TimeSeries series, TimeLevel timeLevel){

..}

```

It is also easier to use and type safe:
`DeMarkPivotPointIndicator d = new DeMarkPivotPointIndicator(series, TimeLevel.week);`

instead of

```
// can accept any int-number, which should not be allowed
DeMarkPivotPointIndicator d = new DeMarkPivotPointIndicator(series, 7);
```